internal/store/memory_backend.go

```diff
@@ -1,40 +1,39 @@
-package memory
+package store
 
 import (
 	"context"
 	"sync"
 
 	"github.com/hashicorp/consul-api-gateway/internal/core"
-	"github.com/hashicorp/consul-api-gateway/internal/store"
 )
 
-type Backend struct {
+type memoryBackend struct {
```

Comment:
Follows the naming pattern established by `consulBackend` in `consul_backend.go`